### ISSUE TITLE
How to caculate distance with the vectors?

### ISSUE THREAD

**Issue by mmpang** [cnclabs/smore#46]
I got the user vector and item vector with nemf(collaborative similarity embedding)，which distance should I use to do ann search，inner product (dot product) or cosine distance or Euclidean distance ?

**Comment by chihming** [comment]
both dot product and eu-distance are okay

**Comment by mmpang** [comment]
@chihming Thanks for your reply. 
Another question, in a recommander system, can i do u2i or i2i retrieval with the vectors directly ?

**Comment by chihming** [comment]
yes, such vector-based retrieval is able to achieve a reasonable performance
